server/src/server.ts

```diff
@@ -224,13 +230,33 @@ connection.onRequest("powerquery/renameIdentifier", async (params: RenameIdentif
         return await analysis.getRenameEdits(params.newName);
     } catch (error) {
         connection.console.error(
-            `on powerquery/renameIdentifier error ${formatError(ErrorUtils.assertAsError(error))}`,
+            `on powerquery/renameIdentifier error ${ErrorUtils.formatError(ErrorUtils.assertAsError(error))}`,
         );
 
         return [];
     }
 });
 
+connection.onRequest("powerquery/semanticTokens", async (params: SemanticTokenParams) => {
```

Comment:
It's interesting we have to add this as a custom request handler. I would have expected it to be built into the LSP client/server library - it looks like there is a handler defined in the lib. https://github.com/microsoft/vscode-languageserver-node/blob/081ba82a0e49eebec2d1b7d93751290cc1128104/server/src/common/semanticTokens.ts

Is that only available in the preview version of the library?

Reply:
The examples I saw did it this way. I also tried implementing it locally but things break on the vscode dependency.